05/DB.test.js

```diff
@@ -1 +1,110 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('DB', () => {
+  let db;
+
+  beforeEach(() => {
+    db = new DB();
+  });
+
+  afterEach(() => {
+    db = null;
+  });
+
+  describe('Insert method', () => {
+    it('Should insert data and resolve with the inserted data', async () => {
+      const data = { id: 1, name: 'John' };
+      const result = await db.insert(data);
+      expect(result).toEqual(data);
+    });
+
+    it('Should generate an ID and resolve with the inserted data if ID is not provided', async () => {
+      const data = { name: 'Jane' };
+      const result = await db.insert(data);
+      expect(result.id).toBeDefined();
+      expect(result.name).toBe('Jane');
+    });
+
+    it('Should reject with an error if data has a non-numeric ID', async () => {
+      const data = { id: 'invalid', name: 'John' };
+      await expect(db.insert(data)).rejects.toMatch('ID can be only number!');
```

Comment:
Zobacz, że dzięki odpowiednim komunikatom wiesz co się dzieje w kodzie i czy się dzieje to czego oczekujesz ;)